ocean_lib/ocean/ocean_assets.py

```diff
@@ -928,3 +731,127 @@ def _start_or_reuse_order_based_on_initialize_response(
+        self.uri = uri or "https://oceanprotocol.com/nft/"
+        self.transferable = transferable or True
+        self.owner = owner
+
+    def publish_from(self, config_dict, wallet):
+        address = get_address_of_type(config_dict, DataNFTFactoryContract.CONTRACT_NAME)
+        data_nft_factory = DataNFTFactoryContract(config_dict, address)
+
+        receipt = data_nft_factory.deployERC721Contract(
+            self.name,
+            self.symbol,
+            self.template_index,
+            self.additional_metadata_updater,
+            self.additional_datatoken_deployer,
+            self.uri,
+            self.transferable,
+            self.owner or wallet.address,
+            {"from": wallet},
+        )
+
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                message=".*Event log does not contain enough topics for the given ABI.*",
+            )
+            registered_event = receipt.events["NFTCreated"]
+
+        data_nft_address = registered_event["newTokenAddress"]
+        return DataNFT(config_dict, data_nft_address)
+
+
+class DatatokenArguments:
+    def __init__(
+        self,
+        name: str,
+        symbol: str,
+        template_index: Optional[int] = 1,
+        minter: Optional[str] = None,
+        fee_manager: Optional[str] = None,
+        publish_market_order_fee_address: Optional[str] = None,
+        publish_market_order_fee_token: Optional[str] = None,
+        publish_market_order_fee_amount: Optional[int] = 0,
+        bytess: Optional[List[bytes]] = None,
+        services: Optional[list] = None,
+        files: Optional[List[FilesType]] = None,
+        consumer_parameters: Optional[List[Dict[str, Any]]] = None,
+    ):
+        self.name = name
+        self.symbol = symbol
+        self.template_index = template_index
+        self.minter = minter
+        self.fee_manager = fee_manager
+        self.publish_market_order_fee_address = publish_market_order_fee_address
+        self.publish_market_order_fee_token = publish_market_order_fee_token
+        self.publish_market_order_fee_amount = publish_market_order_fee_amount
+        self.bytess = bytess or [b""]
+        self.services = services
+        self.files = files
+        self.consumer_parameters = consumer_parameters
+
+    def create_from_arguments(self, config_dict, data_nft, wallet):
+        OCEAN_address = get_ocean_token_address(config_dict)
+        temp_dt = data_nft.create_datatoken(
+            name=self.name,
+            symbol=self.symbol,
+            template_index=self.template_index,
+            minter=self.minter or wallet.address,
+            fee_manager=self.fee_manager or wallet.address,
+            publish_market_order_fee_address=self.publish_market_order_fee_address
+            or ZERO_ADDRESS,
+            publish_market_order_fee_token=self.publish_market_order_fee_token
+            or OCEAN_address,
```

Comment:
OCEAN address can not be moved since it depends on the config dict. I wouldn't add it to the constructor, I'd rather keep it alongside wallet, only when publishing

Reply:
Right. Makes sense. OK then.